src/map.rs

```diff
@@ -432,6 +432,31 @@ fn example() -> Result<(), Box<dyn std::error::Error>> {
     pub fn into_fst(self) -> raw::Fst<D> {
         self.0
     }
+
+    /// Maps the underlying data of the fst Map to another data type.
+    ///
+    /// # Example
+    ///
+    /// This example shows that you can map an fst Map based on a `Vec<u8>`
+    /// into an fst Map based on a `Cow<[u8]>`, it can also work with a
+    /// reference counted type (e.g. `Arc`, `Rc`)
+    ///
+    /// ```rust
+    /// use std::borrow::Cow;
+    /// use fst::Map;
+    ///
+    /// let map: Map<Vec<u8>> = Map::from_iter([("hello", 12), ("world", 42)].iter().cloned()).unwrap();
```

Comment:
Could you make sure all lines, including comments, are wrapped to 79 columns inclusive? Thanks!

In this case, I'd probably write it like this (at least until rustfmt is able to format doc comments):

```rust
let map: Map<Vec<u8>> = Map::from_iter(
    [("hello", 12), ("world", 42)].iter().cloned(),
).unwrap();
```